### PR TITLE
[Bug Fix ] Update ReplaceTool.cs to spawn prefabs. [New Item] Created SwapPrefabs alternative

### PR DESCRIPTION
**Pull request by jandd661**
Is this PR linked to an issue? If so please link it here, if not please create one first, then link it.  

https://github.com/UnityTechnologies/open-project-1/pull/214

https://open.codecks.io/unity-open-project-1/decks/undefined-/card/18v-replacement-tool-instantiates-objects-not-prefabs

Is there a forum thread linked to this issue? If so please link it here.  
No

How did you resolve this issue?
fixed Instantiation method  

How can it be verified that the issue has actually been resolved?  
Yes

### CODE REVIEW

**Comment by jandd661** [comment]
Currently working on an improved version of this system. It will be ready in a day or two.

**Comment by jandd661** [comment]
This tool searches the scene for prefabs that derived from the selected prefab from the project view. Provides a preview of these objects and notifies the user when targeted scene objects have non-prefab child objects.

**Card:** 
https://open.codecks.io/unity-open-project-1/decks/undefined-/card/17s-editor-tool-to-replace-gameobjects-with-prefabs

**Demo:**
https://youtu.be/VsFT8O7gn0U

File: UOP1_Project/Assets/Scripts/Editor/ReplaceTool.cs [hunk 1]
```diff
@@ -137,12 +137,17 @@ private void ReplaceSelectedObjects(GameObject[] objectToReplace, GameObject rep
 		{
 			var go = objectToReplace[i];
 			Undo.RegisterCompleteObjectUndo(go, "Saving game object state");
-			var inst = Instantiate(replaceObject, go.transform.position, go.transform.rotation, go.transform.parent);
-			inst.transform.localScale = go.transform.localScale;
-			Undo.RegisterCreatedObjectUndo(inst, "Replacement creation.");
+			//var inst = Instantiate(replaceObject, go.transform.position, go.transform.rotation, go.transform.parent);
```
Comment (davejrodriguez) [inline UOP1_Project/Assets/Scripts/Editor/ReplaceTool.cs:140]: Not really necessary to keep this around as a comment if it's been replaced.

File: UOP1_Project/Assets/Scripts/EditorTools/SwapPrefabs/Editor/SwapPrefabs.cs [hunk 2]
```diff
@@ -0,0 +1,584 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor;
+
+namespace ChopChop.EditorTools.jandd661
```
Comment (davejrodriguez) [inline UOP1_Project/Assets/Scripts/EditorTools/SwapPrefabs/Editor/SwapPrefabs.cs:5]: We've been using the format of UOP1.[SystemName]. The conventions document should probably be updated to include this.

File: UOP1_Project/Assets/Scripts/EditorTools/SwapPrefabs/Editor/SwapPrefabs.cs [hunk 3]
```diff
@@ -0,0 +1,584 @@
...
+{
+    public class SwapPrefabs : EditorWindow
```
Comment (davejrodriguez) [inline UOP1_Project/Assets/Scripts/EditorTools/SwapPrefabs/Editor/SwapPrefabs.cs:7]: So this is another replace tool? Perhaps this file should be in a separate PR so that this one just focuses on addressing the bug in the original tool?

Comment (jandd661) [reply UOP1_Project/Assets/Scripts/EditorTools/SwapPrefabs/Editor/SwapPrefabs.cs:7]: I was trying to do that but, Making a new PR, it just updates this one. How do you start a seperate PR? Do I need to remove my fork? I have been looking around how to do this but it remains unclear.

File: UOP1_Project/Assets/Scripts/EditorTools/SwapPrefabs/Editor/SwapPrefabs.cs (hunk 2 above)
Comment (jandd661) [reply UOP1_Project/Assets/Scripts/EditorTools/SwapPrefabs/Editor/SwapPrefabs.cs:5]: Yes it should.

File: UOP1_Project/Assets/Scripts/Editor/ReplaceTool.cs (hunk 1 above)
Comment (jandd661) [reply UOP1_Project/Assets/Scripts/Editor/ReplaceTool.cs:140]: agreed

**Comment by jandd661** [comment]
PR https://github.com/UnityTechnologies/open-project-1/pull/223 is a much better fix/replacement to the current ReplaceTool and this fix. That makes the original intent of this PR obsolete. I'm going to close this PR and start a separate one for the SwapPrefabs tool.
